src/aap_eda/api/serializers/source.py

```diff
@@ -0,0 +1,91 @@
+#  Copyright 2023 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import yaml
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+
+from aap_eda.core import models
+
+
+class SourceSerializer(serializers.ModelSerializer):
+    decision_environment_id = serializers.IntegerField(
+        required=False,
+        allow_null=True,
+    )
+
+    user = serializers.HiddenField(default=serializers.CurrentUserDefault())
+
+    class Meta:
+        model = models.Source
+        read_only_fields = [
+            "id",
+            "uuid",
+            "created_at",
+            "modified_at",
+        ]
+        fields = [
+            "name",
+            "args",
+            "type",
+            "decision_environment_id",
+            "user",
+            *read_only_fields,
+        ]
+
+    def validate_args(self, value):
+        if value:
+            try:
+                parsed_args = yaml.safe_load(value)
+            except yaml.YAMLError:
+                raise ValidationError("Invalid YAML format for 'args'")
+
+            if not isinstance(parsed_args, dict):
+                raise ValidationError(
+                    "The 'args' field must be a YAML object (dictionary)"
+                )
+
+            return parsed_args
+        return value
+
+
+class SourceOutSerializer(serializers.ModelSerializer):
+    user = serializers.SerializerMethodField()
+    args = serializers.SerializerMethodField()
```

Comment:
When we override a field with a MethodField we need as well to define it for the openapi schema. Example: https://github.com/ansible/eda-server/blob/8cecffdd5856f4deff062212afb51e0132dda77f/src/aap_eda/api/serializers/rulebook.py#L253